File: HW/HW5_繼承/HW5_繼承/Circle.cpp
#include "Circle.h"

Circle::Circle()
{
	radius = 1;
}

Circle::Circle(double radius, string color, bool filled)
{
	this->radius = radius;
	setColor(color);
	setFilled(filled);
}

void Circle::setRadius(double radius)
{
	this->radius = radius;
}

double Circle::getRadius()
{
	return radius;
}

string Circle::toString()
{
	stringstream ss;
	ss << "circle, " << radius;
	return ss.str();
}

double Circle::getArea()
{
	return pow(radius, 2) * PI;
}

double Circle::getPerimeter()
{
	return 2 * radius * PI;
}

File: python/GeometricObject_Inheritance/PythonApplication1/Circle.py
from GeometricObject import GeometricObject
import math

class Circle(GeometricObject):

    def __init__(self, radius, color , filled ):
        super().__init__(color, filled)
        self.radius = radius

    def setRadius(self, radius):
        self.radius = radius

    def getRadius(self):
        return self.radius

    def getArea(self):
        return self.radius * self.radius * math.pi

    def getDiameter(self):
        return (self.radius * 2)

    def getPerimeter(self):
        return self.radius * 2 * math.pi
